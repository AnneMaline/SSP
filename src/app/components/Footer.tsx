const Footer = () => {
  return (
    <footer className="bg-gray-800 text-white py-10">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
        <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-10">
          {/* Info Section */}
          <div>
            <h3 className="text-lg font-bold mb-4">Info</h3>
            <ul className="space-y-2">
              <li>
                <a href="/about" className="hover:underline">
                  About Us
                </a>
              </li>
              <li>
                <a href="/privacy" className="hover:underline">
                  Privacy Policy
                </a>
              </li>
              <li>
                <a href="/terms" className="hover:underline">
                  Terms & Conditions
                </a>
              </li>
            </ul>
          </div>

          {/* Organization Section */}
          <div>
            <h3 className="text-lg font-bold mb-4">Organization</h3>
            <ul className="space-y-2">
              <li>
                <a href="/team" className="hover:underline">
                  Our Team
                </a>
              </li>
              <li>
                <a href="/careers" className="hover:underline">
                  Careers
                </a>
              </li>
              <li>
                <a href="/news" className="hover:underline">
                  Latest News
                </a>
              </li>
            </ul>
          </div>

          {/* Contact Information Section */}
          <div>
            <h3 className="text-lg font-bold mb-4">Contact</h3>
            <ul className="space-y-2">
              <li>
                <a href="mailto:info@example.com" className="hover:underline">
                  info@example.com
                </a>
              </li>
              <li>
                <a href="tel:+1234567890" className="hover:underline">
                  +123 456 7890
                </a>
              </li>
              <li>
                <a
                  href="https://www.example.com/contact"
                  className="hover:underline"
                >
                  Contact Form
                </a>
              </li>
            </ul>
          </div>
        </div>

        {/* Footer Bottom */}
        <div className="mt-10 text-center text-sm">
          <p>&copy; 2024 Your Organization. All rights reserved.</p>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
