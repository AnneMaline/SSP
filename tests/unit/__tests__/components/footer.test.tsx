import { render, screen } from "@testing-library/react";
import Footer from "../../../../src/components/Footer";

describe("Footer", () => {
  it("renders correctly", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  // Update when the code is not a template!!!
  it("should render all sections in the footer", () => {
    render(<Footer />);

    // Check if the sections are present
    expect(screen.getByText("Info")).toBeInTheDocument();
    expect(screen.getByText("Organization")).toBeInTheDocument();
    expect(screen.getByText("Contact")).toBeInTheDocument();
  });

  it("should render all links in the footer", () => {
    render(<Footer />);

    // Check if the links are present
    expect(screen.getByText("About Us")).toHaveAttribute("href", "/about");
    expect(screen.getByText("Privacy Policy")).toHaveAttribute(
      "href",
      "/privacy"
    );
    expect(screen.getByText("Terms & Conditions")).toHaveAttribute(
      "href",
      "/terms"
    );
    expect(screen.getByText("Our Team")).toHaveAttribute("href", "/team");
    expect(screen.getByText("Careers")).toHaveAttribute("href", "/careers");
    expect(screen.getByText("Latest News")).toHaveAttribute("href", "/news");
    expect(screen.getByText("info@example.com")).toHaveAttribute(
      "href",
      "mailto:info@example.com"
    );
    expect(screen.getByText("+123 456 7890")).toHaveAttribute(
      "href",
      "tel:+1234567890"
    );
    expect(screen.getByText("Contact Form")).toHaveAttribute(
      "href",
      "https://www.example.com/contact"
    );
  });

  it("should render the footer bottom text", () => {
    render(<Footer />);

    // Check if the footer bottom text is present
    expect(
      screen.getByText("Â© 2024 Your Organization. All rights reserved.")
    ).toBeInTheDocument();
  });
});
