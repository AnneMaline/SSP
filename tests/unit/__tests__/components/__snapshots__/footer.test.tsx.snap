// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div>
  <footer
    class="bg-gray-800 text-white py-10"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-10"
      >
        <div>
          <h3
            class="text-lg font-bold mb-4"
          >
            Info
          </h3>
          <ul
            class="space-y-2"
          >
            <li>
              <a
                class="hover:underline"
                href="/about"
              >
                About Us
              </a>
            </li>
            <li>
              <a
                class="hover:underline"
                href="/privacy"
              >
                Privacy Policy
              </a>
            </li>
            <li>
              <a
                class="hover:underline"
                href="/terms"
              >
                Terms & Conditions
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h3
            class="text-lg font-bold mb-4"
          >
            Organization
          </h3>
          <ul
            class="space-y-2"
          >
            <li>
              <a
                class="hover:underline"
                href="/team"
              >
                Our Team
              </a>
            </li>
            <li>
              <a
                class="hover:underline"
                href="/careers"
              >
                Careers
              </a>
            </li>
            <li>
              <a
                class="hover:underline"
                href="/news"
              >
                Latest News
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h3
            class="text-lg font-bold mb-4"
          >
            Contact
          </h3>
          <ul
            class="space-y-2"
          >
            <li>
              <a
                class="hover:underline"
                href="mailto:info@example.com"
              >
                info@example.com
              </a>
            </li>
            <li>
              <a
                class="hover:underline"
                href="tel:+1234567890"
              >
                +123 456 7890
              </a>
            </li>
            <li>
              <a
                class="hover:underline"
                href="https://www.example.com/contact"
              >
                Contact Form
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="mt-10 text-center text-sm"
      >
        <p>
          Â© 2024 Your Organization. All rights reserved.
        </p>
      </div>
    </div>
  </footer>
</div>
`;
